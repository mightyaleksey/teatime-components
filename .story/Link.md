## Preview


### theme action

```
<div>
  <Link disabled theme='action'>Link</Link>
  {' '}size s
  <Link size='s' theme='action'>Link</Link>
  {' '}size m
  <Link size='m' theme='action'>Link</Link>
  {' '}size l
  <Link size='l' theme='action'>Link</Link>
</div>
```


### theme link

```
<div>
  <Link disabled theme='link'>Link</Link>
  {' '}size s
  <Link size='s' theme='link'>Link</Link>
  {' '}size m
  <Link size='m' theme='link'>Link</Link>
  {' '}size l
  <Link size='l' theme='link'>Link</Link>
</div>
```


### theme normal

```
<div>
  <Link disabled theme='normal'>Link</Link>
  {' '}size s
  <Link size='s' theme='normal'>Link</Link>
  {' '}size m
  <Link size='m' theme='normal'>Link</Link>
  {' '}size l
  <Link size='l' theme='normal'>Link</Link>
</div>
```
