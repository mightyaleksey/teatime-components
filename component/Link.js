'use strict';

const {Component, PropTypes} = require('react');
const {omitNonStandardAttrs, genericName} = require('../lib/util');
const {noop} = require('../lib/dash');
const React = require('react');
const cc = require('classnames');

const cssModules = {
  'action-l': require('../style/link/link-action-l.css'),
  'action-m': require('../style/link/link-action-m.css'),
  'action-s': require('../style/link/link-action-s.css'),
  'link-l': require('../style/link/link-l.css'),
  'link-m': require('../style/link/link-m.css'),
  'link-s': require('../style/link/link-s.css'),
  'normal-l': require('../style/link/link-normal-l.css'),
  'normal-m': require('../style/link/link-normal-m.css'),
  'normal-s': require('../style/link/link-normal-s.css'),
};

class Link extends Component {
  css = tokenName => genericName(this.props, tokenName)

  focus = noop

  render() {
    const {
      children,
      className,
      download,
      href,
      id,
      target,
      type,
    } = this.props;

    const {css} = this;

    return (
      <a
        {...omitNonStandardAttrs(this.props)}
        className={cc(css('control'), className)}
        download={download}
        href={href}
        id={id}
        target={target}
        type={type}>
        {children}
      </a>
    );
  }
}

Link.defaultProps = {
  size: 'm',
  styles: cssModules,
  theme: 'link',
};

Link.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  download: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
  ]),
  href: PropTypes.string,
  id: PropTypes.string,
  size: PropTypes.oneOf([
    'l',
    'm',
    's',
  ]),
  styles: PropTypes.object,
  target: PropTypes.string,
  theme: PropTypes.oneOf([
    'action',
    'link',
    'normal',
  ]),
  type: PropTypes.string,
};

module.exports = Link;
