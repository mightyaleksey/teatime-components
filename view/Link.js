'use strict';

const { Component, PropTypes } = require('react');
const { styleName } = require('../tool/className');
const { sanitizeProps } = require('../tool/component');
const React = require('react');

class Link extends Component {
  render() {
    return (
      <a
        {...sanitizeProps(['theme', 'styles', 'styleName'], ...this.props)}
        className={styleName(this.props)}/>
    );
  }
}

Link.defaultProps = {
  styleName: 'control',
  styles: {},
};

Link.propTypes = {
  href: PropTypes.string.isRequired,
  styleName: PropTypes.string,
  styles: PropTypes.shape({
    control: PropTypes.string,
  }),
};

module.exports = Link;
