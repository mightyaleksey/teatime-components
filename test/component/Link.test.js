import Link from '../../component/Link';
import React from 'react';
import renderer from 'react-test-renderer';

test('Link renders correctly', () => {
  const tree = renderer.create(
    <Link
      className='outer'
      download={true}
      href='#'
      id='links id'
      target='_blank'
      type='video/mp4'>
      Link Content
    </Link>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
