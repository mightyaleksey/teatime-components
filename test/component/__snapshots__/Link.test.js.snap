exports[`test Link renders correctly 1`] = `
<a
  className="control outer"
  download={true}
  href="#"
  id="links id"
  target="_blank"
  type="video/mp4">
  Link Content
</a>
`;
